extension/src/popup/views/AccountHistory/index.tsx

```diff
@@ -88,18 +89,20 @@ export const AccountHistory = () => {
   const stellarExpertUrl = getStellarExpertUrl(networkDetails);
 
   // differentiate between if data is still loading and if no account history results came back from Horizon
+  const shouldLoadToken =
+    isExperimentalModeEnabled || networkDetails.network === NETWORKS.FUTURENET;
```

Comment:
isn't this more dependent on `isExperimentalModeEnabled`? If that flag is true, `soroban-client` will be used, but it will use `stellar-sdk` if it's false.

Also, since we're going live on Testnet in a few weeks, maybe we should just key this off of `isExperimentalModeEnabled`. I'm guessing the Futurenet check is to make sure the data exists (or something like that). Maybe we should gracefully fail if the data isn't found on network

Reply:
i'm not sure that I understand, how does this toggle the sdk? AFAIk this toggles wether we fetch and add soroban tokens to the account history. The network check is needed because you can be in experimental mode while not being on Futurenet, so as currently implemented it incorrectly shows tokens that are not for the network when on Testnet.

Reply:
Oh actually - you're right. We're just grabbing the Horizon entry, so we can just use `stellar-sdk` here. I was thinking of other places in the app where we do: `const SDK = isExperimentalModeEnabled ? SorobanSdk : StellarSdk;`. You can ignore that first part 😛

Reply:
but still I think also the second part we cannot do right now because the tokens as we store them today have no relation to any network(we should address this before testnet) so when on Testnet but with experimental mode toggled on, you incorrectly see tokens that are on Futurenet without the network check.

Reply:
I guess for the second part: I think instead of checking for Futurenet explicitly here, we should adjust the network url that `SorobanContext` is using.  On closer inspection, looks like `SorobanContext` is hardcoded to use Futurenet. So, in `getTokenBalances` on ln 162, it's always checking Fututenet. 

We should fix that in SorobanContext (but maybe we just do that in a later PR)

This is mostly so when Soroban is running on both Futurenet and Testnet, we have fewer places in the code where we have explicit checks for Futurenet that we have to manually remove

Reply:
We must have been writing this at the same time. Yeah we should def be able to configure the SorobanContext for other networks before Testnet support is live. We still can't set the network in the SorobanContext to whatever the current network is here to fix this bug because a) the saved tokens have no relation to a network today, b) There is still only futurenet support so this is incorrect today.

Reply:
There's def a bigger refactor here to support future networks but this is just meant to address the live bug where we show users the wrong tokens when in experimental mode and not on Futurenet. In order to fix this to work for other networks, we also need to refactor how we store tokens to include network details, then change all token related checks to instead map to tokens that are saved for that network.

Reply:
Cool - yeah, let's just leave this is as-is then, and we can tackle the network stuff later on

Reply:
but good call out as I hadn't really considered before this that there will be significant work to support tokens on other networks, and it will include the always hairy work of migrating storage schemas.